src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -115,53 +117,42 @@ public static void CheckHandle (IntPtr jnienv)
 		static unsafe void RegisterJniNatives (IntPtr typeName_ptr, int typeName_len, IntPtr jniClass, IntPtr methods_ptr, int methods_len)
 		{
 			string typeName = new string ((char*) typeName_ptr, 0, typeName_len);
-
-			var __start = new DateTime ();
-			if (Logger.LogTiming) {
-				__start = DateTime.UtcNow;
-				Logger.Log (LogLevel.Info,
-						"monodroid-timing",
-						"JNIEnv.RegisterJniNatives (\"" + typeName + "\", 0x" + jniClass.ToString ("x") + ") start: " + (__start - new DateTime (1970, 1, 1)).TotalMilliseconds);
-			}
-
 			Type type = Type.GetType (typeName);
 			if (type == null) {
 				Logger.Log (LogLevel.Error, "MonoDroid",
-						"Could not load type '" + typeName + "'. Skipping JNI registration of type '" + 
+						"Could not load type '" + typeName + "'. Skipping JNI registration of type '" +
 						Java.Interop.TypeManager.GetClassName (jniClass) + "'.");
 				return;
 			}
 
-			var className = Java.Interop.TypeManager.GetClassName (jniClass);
+			var className = Java.Interop.TypeManager.GetClassName (jniClass); // TODO: <- replace contents of GetClassName with p/invokes
 			TypeManager.RegisterType (className, type);
 
 			JniType jniType = null;
 			JniType.GetCachedJniType (ref jniType, className);
 
 			androidRuntime.TypeManager.RegisterNativeMembers (jniType, type, methods_ptr == IntPtr.Zero ? null : new string ((char*) methods_ptr, 0, methods_len));
-
-			if (Logger.LogTiming) {
-				var __end = DateTime.UtcNow;
-				Logger.Log (LogLevel.Info,
-						"monodroid-timing",
-						"JNIEnv.RegisterJniNatives total time: " + (__end - new DateTime (1970, 1, 1)).TotalMilliseconds + " [elapsed: " + (__end - __start).TotalMilliseconds + " ms]");
-			}
 		}
 
+		[DllImport ("__Internal")]
+		extern static void monodroid_log (LogLevel level, LogCategories category, string message);
```

Comment:
At *this* point in time, it's "weird" to have this and the following methods declared when they're not actually used from C#.  I imagine this will change in the future, but for *now*, this is weird.

Reply:
The other two are used, this one was used instead of passing a string to `monodroid_timing_{start,end}` but it will still come useful IMO (I think it might be a good idea to replace `Logger.Log` calls with this in the code we ship)